parts/file.go

```diff
@@ -0,0 +1,162 @@
+package parts
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/apache/arrow/go/v14/arrow"
+	"github.com/parquet-go/parquet-go"
+
+	"github.com/polarsignals/frostdb/dynparquet"
+)
+
+// fileParquetPart implements the Part interface. Is is backed by a Parquet file.
+type fileParquetPart struct {
+	basePart
+
+	rdr     io.ReaderAt
+	size    int64
+	release func()
+}
+
+func NewFileParquetPart(tx uint64, rdr io.ReaderAt, size int64, release func(), options ...Option) Part {
```

Comment:
Below I comment on sharing some code with the parquet part but thinking more about this I think we could just remove this new file part and have the existing parquet part use a disk-backed reader at, right? It's basically up to whoever creates the part to create the reader at, but the part itself is agnostic to the backing storage. I think we just need to add the release functionality.

Reply:
So initially it made more sense as when I wrote this I had the `fileParquetPart` handling the file(s) themselves. But as it evolved it definitely makes less sense. 

There are some subtle differences between the two still that make me wonder if it's best to merge them. Such as the file part opens the file each time to avoid holding pages in memory, I'm wondering if that behavior makes sense for an already in memory part.

Reply:
Hmm, I didn't notice that. Are we sure we want to open the file every time? This also includes re-reading parts of the file that would have already been read by a previous open.

Reply:
I don't know yet. Need more real world data on that. Since the goal was to limit the amount of memory to store parts it made more sense to me to not hold onto any of that memory. 

Since in theory the file will only ever be opened on query and compaction.

Reply:
How much memory do we actually use with just keeping an open file around? I would assume variable-sized buffers are only used when reading/decoding. We would already be reducing the memory used just for the backing storage with the in-mem parquet part.

Reply:
You're right, tested it out and there was no noticeable overhead. I've removed the file part and replaced the usage with the normal Parquet part.